src/rust/engine/fs/src/directory_tests.rs

```diff
@@ -0,0 +1,372 @@
+// Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::collections::HashMap;
+
+use crate::directory::{DigestTrie, Entry, Name, TypedPath};
+use crate::MAX_LINK_DEPTH;
+use hashing::EMPTY_DIGEST;
+use std::path::{Path, PathBuf};
+
```

Comment:
Thanks for doing this... I'm lightly embarassed not to have had rust side tests for `walk`/`entry` already... but on the other hand, it used to be a _lot_ simpler, heh.

Reply:
Having these tests is the only way I survived honestly. Turnaround time for changes went from minutes to seconds, and the ability to log/debug easily was invaluable.